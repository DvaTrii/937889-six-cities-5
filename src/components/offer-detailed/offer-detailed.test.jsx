import React from "react";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import OfferDetailed from "./offer-detailed.tsx";

const nearOffers = [
  {
    id: 101,
    city: {
      name: `Amsterdam`,
      coordinates: { // был [33, 33]
        latitude: 52.38333,
        longitude: 4.9
      }
    },
    coordinates: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198
    },
    previewImage: `img/apartment-02.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`],
    isPremium: true,
    isBookmark: true,
    price: 100,
    title: `Wood and stone place`,
    type: `House`,
    rating: 5,
    description: `Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium`,
    bedroomsMax: `4 Bedrooms`,
    guestsMax: `Max 5 adults`,
    amenities: [`Wi-Fi`, `Heating`, `Kitchen`, `Parking`, `Flowers`, `Dishwasher`, `Towels`, `TV`],
    hostInfo: {
      avatar: `img/avatar-max.jpg`,
      name: `Ross`,
      isSuper: true,
    },
  },
  {
    id: 102,
    city: {
      name: `Amsterdam`,
      coordinates: { // был [33, 33]
        latitude: 52.38333,
        longitude: 4.9
      }
    },
    coordinates: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198
    },
    previewImage: `img/apartment-02.jpg`,
    pictures: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    isPremium: false,
    isBookmark: true,
    price: 80,
    title: `Nice, cozy, warm big bed apartment`,
    type: `Hotel`,
    rating: 4,
    description: `Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit`,
    bedroomsMax: `4 Bedrooms`,
    guestsMax: `Max 5 adults`,
    amenities: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Dishwasher`, `Towels`, `TV`],
    hostInfo: {
      avatar: `img/avatar-max.jpg`,
      name: `Chandler`,
      isSuper: true,
    },
  },
  {
    id: 103,
    city: {
      name: `Amsterdam`,
      coordinates: { // был [33, 33]
        latitude: 52.38333,
        longitude: 4.9
      }
    },
    coordinates: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198
    },
    previewImage: `img/apartment-02.jpg`,
    pictures: [`img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`],
    isPremium: false,
    isBookmark: false,
    price: 120,
    title: `Canal View Prinsengracht`,
    type: `Hotel`,
    rating: 3,
    description: `Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam`,
    bedroomsMax: `1 Bedroom`,
    guestsMax: `Max 2 adults`,
    amenities: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Dishwasher`, `Towels`, `TV`],
    hostInfo: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Monica`,
      isSuper: false,
    },
  },
];
const reviews = [
  {
    id: 300,
    date: 1605357744252,
    userInfo: {
      avatar: `img/avatar-max.jpg`,
      name: `Tom`
    },
    text: `Lorem ipsum dolor sit et dolore magna aliqua. Ut enim ad minim veniam`,
    rating: 4
  },
  {
    id: 333,
    date: 1605357744253,
    userInfo: {
      avatar: `img/avatar-max.jpg`,
      name: `Bob`
    },
    text: `Lorem ipsum dolor sir incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam`,
    rating: 3
  },
];
const offer = {
  id: 101,
  city: {
    name: `Amsterdam`,
    coordinates: { // был [33, 33]
      latitude: 52.38333,
      longitude: 4.9
    }
  },
  coordinates: {
    latitude: 52.3909553943508,
    longitude: 4.85309666406198
  },
  previewImage: `img/apartment-02.jpg`,
  pictures: [`img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`],
  isPremium: true,
  isBookmark: true,
  price: 100,
  title: `Wood and stone place`,
  type: `House`,
  rating: 5,
  description: `Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium`,
  bedroomsMax: 4,
  guestsMax: 5,
  amenities: [`Wi-Fi`, `Heating`, `Kitchen`, `Parking`, `Flowers`, `Dishwasher`, `Towels`, `TV`],
  hostInfo: {
    avatar: `img/avatar-max.jpg`,
    name: `Ross`,
    isSuper: true,
  },
};
const status = true;

const mockStore = configureStore([]);

it(`Should OfferDetailed render properly`, () => {

  const store = mockStore({
    APP: {hoveredOfferId: -1},
    DATA: {offerById: offer}
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <OfferDetailed
              offer={offer}
              reviews={reviews}
              nearOffers={nearOffers}
              isLoadedNearOffers={status}
              isLoaded={status}
              authorizationStatus={status}
            />
          </BrowserRouter>
        </Provider>, {
          createNodeMock: () => document.createElement(`div`)
        }
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
