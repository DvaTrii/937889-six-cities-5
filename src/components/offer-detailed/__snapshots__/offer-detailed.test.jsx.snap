// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should OfferDetailed render properly 1`] = `
<section
  className="property"
>
  <div
    className="property__gallery-container container"
  >
    <div
      className="property__gallery"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="property__container container"
  >
    <div
      className="property__wrapper"
    >
      <div
        className="property__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="property__name-wrapper"
      >
        <h1
          className="property__name"
        >
          Wood and stone place
        </h1>
        <button
          className="place-card__bookmark-button
              place-card__bookmark-button--active
              button"
          type="button"
        >
          <svg
            className="property__bookmark-icon"
            height="33"
            width="31"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="property__rating rating"
      >
        <div
          className="property__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
        <span
          className="property__rating-value rating__value"
        >
          5
        </span>
      </div>
      <ul
        className="property__features"
      >
        <li
          className="property__feature property__feature--entire"
        >
          House
        </li>
        <li
          className="property__feature property__feature--bedrooms"
        >
          4
           $
          Bedrooms
        </li>
        <li
          className="property__feature property__feature--adults"
        >
          Max 
          5
           adults
        </li>
      </ul>
      <div
        className="property__price"
      >
        <b
          className="property__price-value"
        >
          €
          100
        </b>
        <span
          className="property__price-text"
        >
           night
        </span>
      </div>
      <div
        className="property__inside"
      >
        <h2
          className="property__inside-title"
        >
          What's inside
        </h2>
        <ul
          className="property__inside-list"
        >
          <li
            className="property__inside-item"
          >
            Wi-Fi
          </li>
          <li
            className="property__inside-item"
          >
            Heating
          </li>
          <li
            className="property__inside-item"
          >
            Kitchen
          </li>
          <li
            className="property__inside-item"
          >
            Parking
          </li>
          <li
            className="property__inside-item"
          >
            Flowers
          </li>
          <li
            className="property__inside-item"
          >
            Dishwasher
          </li>
          <li
            className="property__inside-item"
          >
            Towels
          </li>
          <li
            className="property__inside-item"
          >
            TV
          </li>
        </ul>
      </div>
      <div
        className="property__host"
      >
        <h2
          className="property__host-title"
        >
          Meet the host
        </h2>
        <div
          className="property__host-user user"
        >
          <div
            className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
          >
            <img
              alt="Host avatar"
              className="property__avatar user__avatar"
              height="74"
              src="img/avatar-max.jpg"
              width="74"
            />
          </div>
          <span
            className="property__user-name"
          >
            Ross
          </span>
        </div>
        <div
          className="property__description"
        >
          <p
            className="property__text"
          >
            Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium
          </p>
        </div>
      </div>
      <section
        className="property__reviews reviews"
      >
        <h2
          className="reviews__title"
        >
          Reviews · 
          <span
            className="reviews__amount"
          >
            4
          </span>
        </h2>
        <ul
          className="reviews__list"
        >
          <li
            className="reviews__review"
          >
            <div
              className="reviews__user user"
            >
              <div
                className="reviews__avatar-wrapper user__avatar-wrapper"
              >
                <img
                  alt="Reviews avatar"
                  className="reviews__avatar user__avatar"
                  height="54"
                  src="img/avatar-max.jpg"
                  width="54"
                />
              </div>
              <span
                className="reviews__user-name"
              >
                Tom
              </span>
            </div>
            <div
              className="reviews__info"
            >
              <div
                className="reviews__rating rating"
              >
                <div
                  className="reviews__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "80%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <p
                className="reviews__text"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam
              </p>
              <time
                className="reviews__time"
                dateTime="April 2019"
              >
                April 2019
              </time>
            </div>
          </li>
          <li
            className="reviews__review"
          >
            <div
              className="reviews__user user"
            >
              <div
                className="reviews__avatar-wrapper user__avatar-wrapper"
              >
                <img
                  alt="Reviews avatar"
                  className="reviews__avatar user__avatar"
                  height="54"
                  src="img/avatar-max.jpg"
                  width="54"
                />
              </div>
              <span
                className="reviews__user-name"
              >
                Bob
              </span>
            </div>
            <div
              className="reviews__info"
            >
              <div
                className="reviews__rating rating"
              >
                <div
                  className="reviews__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "60%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <p
                className="reviews__text"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam
              </p>
              <time
                className="reviews__time"
                dateTime="undefined"
              />
            </div>
          </li>
          <li
            className="reviews__review"
          >
            <div
              className="reviews__user user"
            >
              <div
                className="reviews__avatar-wrapper user__avatar-wrapper"
              >
                <img
                  alt="Reviews avatar"
                  className="reviews__avatar user__avatar"
                  height="54"
                  src="img/avatar-angelina.jpg"
                  width="54"
                />
              </div>
              <span
                className="reviews__user-name"
              >
                Jess
              </span>
            </div>
            <div
              className="reviews__info"
            >
              <div
                className="reviews__rating rating"
              >
                <div
                  className="reviews__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "40%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <p
                className="reviews__text"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam
              </p>
              <time
                className="reviews__time"
                dateTime="undefined"
              />
            </div>
          </li>
          <li
            className="reviews__review"
          >
            <div
              className="reviews__user user"
            >
              <div
                className="reviews__avatar-wrapper user__avatar-wrapper"
              >
                <img
                  alt="Reviews avatar"
                  className="reviews__avatar user__avatar"
                  height="54"
                  src="img/avatar-angelina.jpg"
                  width="54"
                />
              </div>
              <span
                className="reviews__user-name"
              >
                Mary
              </span>
            </div>
            <div
              className="reviews__info"
            >
              <div
                className="reviews__rating rating"
              >
                <div
                  className="reviews__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "100%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <p
                className="reviews__text"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam
              </p>
              <time
                className="reviews__time"
                dateTime="undefined"
              />
            </div>
          </li>
        </ul>
        <form
          action="#"
          className="reviews__form form"
          method="post"
          onSubmit={[Function]}
        >
          <label
            className="reviews__label form__label"
            htmlFor="review"
          >
            Your review
          </label>
          <div
            className="reviews__rating-form form__rating"
          >
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              id="5-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value="5"
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="5-stars"
              title="perfect"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              id="4-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value="4"
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="4-stars"
              title="good"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              id="3-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value="3"
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="3-stars"
              title="not bad"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              id="1-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value="2"
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="2-stars"
              title="badly"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
            <input
              checked={false}
              className="form__rating-input visually-hidden"
              id="1-stars"
              name="rating"
              onChange={[Function]}
              type="radio"
              value="1"
            />
            <label
              className="reviews__rating-label form__rating-label"
              htmlFor="1-star"
              title="terribly"
            >
              <svg
                className="form__star-image"
                height="33"
                width="37"
              >
                <use
                  xlinkHref="#icon-star"
                />
              </svg>
            </label>
          </div>
          <textarea
            className="reviews__textarea form__textarea"
            id="review"
            name="review"
            onChange={[Function]}
            placeholder="Tell how was your stay, what you like and what can be improved"
            value=""
          />
          <div
            className="reviews__button-wrapper"
          >
            <p
              className="reviews__help"
            >
              To submit review please make sure to set 
              <span
                className="reviews__star"
              >
                rating
              </span>
               and describe your stay with at least 
              <b
                className="reviews__text-amount"
              >
                50 characters
              </b>
              .
            </p>
            <button
              className="reviews__submit form__submit button"
              disabled={true}
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </section>
    </div>
  </div>
  <section
    className="property__map map"
  >
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  </section>
</section>
`;
